profile/profile_test.go

```diff
@@ -912,6 +945,134 @@ func locationHash(s *Sample) string {
 	return tb
 }
 
+func TestHasTag(t *testing.T) {
+	var testcases = []struct {
+		desc       string
+		sample     *Sample
+		key        string
+		value      string
+		wantHasTag bool
+	}{
+		{
+			desc: "empty label does not have tag",
+			sample: &Sample{
```

Comment:
I think you could avoid repeating the Sample and its Location and Value in every test case. You only care about Label field, you could reuse the rest.